stdlib/public/core/Builtin.swift

```diff
@@ -290,6 +290,11 @@ public func _onFastPath() {
   Builtin.onFastPath()
 }
 
+@_transparent
+public func _assume(_ condition: Bool) {
```

Comment:
I think this should be `@usableFromInline @_transparent inline` for now. (I don't recall an immediate use case for it in the overlays, and lacking that there is no reason to make it fully public.)

Reply:
Regarding the name: shouldn't we call attention to the fact this provides no checks whatsoever that the condition holds? `_assume` on its own seems attractively succinct, but this seems like a dangerous tool with a rather narrow use case. IIUC, this isn't a replacement for `_sanityCheck`, for example.

`_uncheckedAssume(_:)` might be a better name. Arguably we should stick an `unsafe` in there too. 

`_blindlyTrust(that:)`? 

`_weHoldThisTruthToBeSelfEvident(_:)`??

Reply:
It's clearly not self-evident since otherwise the compiler would be able to infer it. ;-)

Reply:
`_assume` has the virtue that it maps directly to the LLVM builtin name. That's worth something. I would be ok with `_uncheckedAssume`.

`_iSolemlySwearThatIAmUpToNoGood`.